File: lab2-var23/generator.cpp
// Генератор случайной битовой последовательности на C++
// Создаёт 128-битную последовательность с фиксированным seed и сохраняет в файл

#include <iostream>
#include <fstream>
#include <cstdlib>
#include <string>

using namespace std;

int main() {
    // Устанавливаем seed для генератора случайных чисел
    srand(24);

    string bitStream;

    // Генерируем 128 случайных битов (0 или 1)
    for (int index = 0; index < 128; ++index) {
        bitStream += char('0' + rand() % 2);
    }

    // Открываем файл для записи
    ofstream outputFile("gen_cpp.txt");

    if (!outputFile) {
        cout << "Не удалось создать файл." << endl;
        return 1;
    }

    // Записываем последовательность в файл
    outputFile << bitStream;
    outputFile.close();

    cout << "Двоичная последовательность записана в gen_cpp.txt" << endl;

    return 0;
}

File: lab2-var23/generator.java
// Генератор случайной битовой последовательности на Java
// Создаёт 128-битную последовательность с фиксированным seed и сохраняет в файл

import java.io.FileWriter;
import java.io.PrintWriter;
import java.util.Random;

public class generator {
    public static void main(String[] args) {
        // Создаём генератор случайных чисел с фиксированным seed
        Random rng = new Random(24);
        StringBuilder bitStream = new StringBuilder();

        // Генерируем 128 случайных битов (0 или 1)
        for (int index = 0; index < 128; index++) {
            bitStream.append(rng.nextInt(2));
        }

        // Записываем последовательность в файл с обработкой исключений
        try (PrintWriter writer = new PrintWriter(new FileWriter("gen_java.txt"))) {
            writer.print(bitStream);
            System.out.println("Последовательность успешно записана в gen_java.txt");
        } catch (Exception ex) {
            System.out.println("Ошибка при работе с файлом: " + ex.getMessage());
        }
    }
}